src/refs.c

```diff
@@ -404,7 +427,13 @@ static int reference__create(
 	} else {
 		git_refname_t normalized_target;
 
-		if ((error = reference_normalize_for_repo(normalized_target, repo, symbolic)) < 0)
+		if (git_reference__enable_symbolic_ref_target_validation) {
+			error = reference_normalize_for_repo(normalized_target, repo, symbolic);
+		} else {
+			error = reference_normalize_for_repo_without_validation(normalized_target, repo, symbolic);
+		}
```

Comment:
This is awkward, and we can drop both of these functions by simply passing the `bool` directly:

    error = reference_normalize_for_repo(normalized_target, repo, symbolic, git_reference__enable_symbolic_ref_target_validation);